### PR TITLE
Decoupling Authentication Flow: Mark V

### PR DESCRIPTION
**Pull request by jleandroperez**
### Description:
In this PR we're implementing the delegation pattern, to channel all access to Helpshift via the **WordPressAuthenticationManager** instance (Client App Side).

Needs Review: @nheagy / @ScoutHarris 
Thanks in advance!!

### Scenario: Support
1. Fresh Install WPiOS
2. Open the Login / Signup flow

Verify that pressing the Help button (top right corner) effectively displays the Support UI.

### Scenario: Displaying Helpshift
1. Fresh install WPiOS
2. Press over LogIn
3. Press over `Log in by entering your site address`
4. Press over `Need help finding your site address?`
5. Press over `Need more help?`

Verify that Helpshift shows up onscreen.

### Scenario: Helpshift Push Notifications
It'd be absolutely neat to verify the Helpshift Push Notifications, but i've been unable to get them to work in the development build. 

*However* we can simulate them!!. Please, do the following:

1. Paste this snippet in your `didFinishLaunching` method (AppDelegate):

```
[[NSNotificationCenter defaultCenter] postNotificationName:HelpshiftUnreadCountUpdatedNotification object:nil];
```

2. Edit `WordPressAuthenticationManager` and add / replace the following methods, so that we can hack the Badge Count:

```
    var supportCountOverride: Int?

    @objc
    func helpshiftUnreadCountWasUpdated(_ notification: Foundation.Notification) {
        supportCountOverride = 5
        WordPressAuthenticator.shared.supportBadgeCountWasUpdated()
    }


    var supportBadgeCount: Int {
        if let supportCountOverride = supportCountOverride {
            return supportCountOverride
        }
        return HelpshiftUtils.unreadNotificationCount()
    }
```

3. Fresh install WPiOS
4. Press over Login / Signup
5. Verify that the Badge displays a red `5`

### CODE REVIEW

**Comment by jleandroperez** [comment]
Since this one doesn't change logic itself (it's just a delegate proxy), @frosty  mind taking a look?

Thanks!!

File: WordPress/Classes/ViewRelated/NUX/NUXAbstractViewController.swift [hunk 1]
```diff
@@ -137,14 +137,15 @@ class NUXAbstractViewController: UIViewController, NUXSegueHandler, LoginWithLog
         }
 
         setupHelpButtonAndBadge()
+        refreshBadgeCount()
     }
 
     /// Sets up the help button and the helpshift conversation badge.
```
Comment (frosty) [inline WordPress/Classes/ViewRelated/NUX/NUXAbstractViewController.swift]: Do you want to remove the reference to `Helpshift` here too?

**Comment by jleandroperez** [comment]
@frosty i realize i never replied to your WordPressAuthenticator / WordPressAuthenticationManager Q.

Those two are actually being refactored, and there's *a lot* of stuff coming to `WordPressAuthenticationManager` (it's only that at this stage, only the Helpshift / Support parts made their way thru).

Thanks for the feedback!!!

**Comment by jleandroperez** [comment]
Thank you James!!

**Pull request merged**